vsintegration/src/FSharp.Editor/Navigation/NavigateToSearchService.fs

```diff
@@ -250,14 +254,18 @@ type internal FSharpNavigateToSearchService
                         |> Array.filter (fun x -> x.Name.Length = 1 && String.Equals(x.Name, searchPattern, StringComparison.InvariantCultureIgnoreCase))
                     else
                         [| yield! items |> Array.map (fun items -> items.Find(searchPattern)) |> Array.concat
-                           use patternMatcher = new PatternMatcher(searchPattern, allowFuzzyMatching = true)
+                           let patternMatcherOptions = PatternMatcherCreationOptions(
+                               cultureInfo = CultureInfo.CurrentUICulture,
+                               flags = PatternMatcherCreationFlags.AllowFuzzyMatching
+                           )
+                           let patternMatcher = patternMatcherFactory.CreatePatternMatcher(searchPattern, patternMatcherOptions)
                            yield! items
                                   |> Array.collect (fun item -> item.AllItems)
-                                  |> Array.Parallel.collect (fun x -> 
-                                      patternMatcher.GetMatches(x.LogicalName)
-                                      |> Seq.map (fun pm ->
-                                          NavigateToSearchResult(x, patternMatchKindToNavigateToMatchKind pm.Kind) :> FSharpNavigateToSearchResult)
-                                      |> Seq.toArray) |]
+                                  |> Array.Parallel.choose (fun x -> 
+                                      patternMatcher.TryMatch(x.LogicalName)
+                                      |> Option.ofNullable
+                                      |> Option.map (fun pm ->
+                                          NavigateToSearchResult(x, patternMatchKindToNavigateToMatchKind pm.Kind) :> FSharpNavigateToSearchResult)) |]
 
                 return items |> Array.distinctBy (fun x -> x.NavigableItem.Document.Id, x.NavigableItem.SourceSpan)
```

Comment:
The PatternMatchKind implements IComparable and detects the kind of match, e.g. Exact being the top priority.
=> Wouldn't it make sense to sort all of the results (sort by the "PatternMatch" struct) ?

Reply:
I think the IDE already does this as it integrates results from many sources in the search window.